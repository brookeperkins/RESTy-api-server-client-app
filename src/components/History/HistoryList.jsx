import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
/* eslint-disable max-len */

const HistoryList = ({ histories }) => {
  const historyElements = histories.map(history => (
    <li key={`${history.method}+${history.url}`}>
      <HistoryItem method={history.method} url={history.url} body={history.body} />
    </li>
  ));

  return (
    <ul className={HistoryList}>
      <li>Request History</li>
      {historyElements}
    </ul>
  );
};

HistoryList.propTypes = {
  histories: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    body: PropTypes.string,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default HistoryList;
