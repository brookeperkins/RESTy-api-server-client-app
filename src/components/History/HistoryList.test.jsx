import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';
/* eslint-disable max-len */

describe('HistoryList component', () => {
  afterEach(() => cleanup());

  const testArray = [
    { url: 'https://hey-arnold-api.herokuapp.com/api/v1/characters/random', 
      method: 'GET' },
    { url: 'https://hey-arnold-api.herokuapp.com/api/v1/characters/random',
      method: 'POST' },
    { url: 'https://hey-arnold-api.herokuapp.com/api/v1/characters/random',
      method: 'GET' }
  ];

  it('render HistoryList component', () => {
    const { asFragment } = render(<HistoryList
      histories={testArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
