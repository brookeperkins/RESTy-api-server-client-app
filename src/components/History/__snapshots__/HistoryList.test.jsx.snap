// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component render HistoryList component 1`] = `
<DocumentFragment>
  <ul>
    <li>
      Request History
    </li>
    <li>
      <section>
        <h2>
          GET
        </h2>
        <p>
          https://hey-arnold-api.herokuapp.com/api/v1/characters/random
        </p>
        <p />
      </section>
    </li>
    <li>
      <section>
        <h2>
          POST
        </h2>
        <p>
          https://hey-arnold-api.herokuapp.com/api/v1/characters/random
        </p>
        <p />
      </section>
    </li>
    <li>
      <section>
        <h2>
          GET
        </h2>
        <p>
          https://hey-arnold-api.herokuapp.com/api/v1/characters/random
        </p>
        <p />
      </section>
    </li>
  </ul>
</DocumentFragment>
`;
